lib/atom-io-client.coffee

```diff
@@ -208,11 +208,10 @@ class AtomIoClient
         if err
           error = new Error("Searching for \u201C#{query}\u201D failed.")
           error.stderr = err.message
-          reject error
+          reject(error)
         else
           resolve(
-            body.filter (pkg) -> pkg.releases?.latest?
-                .map ({readme, metadata, downloads, stargazers_count}) ->
-                  Object.assign metadata, {readme, downloads, stargazers_count}
-                .sort (a, b) -> b.downloads - a.downloads
+            body.filter (pack) ->
+              pack.releases?.latest?.map ({readme, metadata, downloads, stargazers_count}) ->
```

Comment:
`.filter` and `.map` need to be on the same level: first filter is done, then map, not one inside other.

Reply:
Ah, I must have touched that line by accident when removing `.sort`.  Then Coffeelint gave me errors and I guess I didn't notice and "fixed" it incorrectly :/.